application/comit_node/src/swap_protocols/rfc003/actions/erc20.rs

```diff
@@ -0,0 +1,64 @@
+use crate::swap_protocols::{
+    ledger::Ethereum,
+    rfc003::{
+        ethereum::{self, Erc20Htlc},
+        state_machine::HtlcParams,
+        Secret,
+    },
+};
+use ethereum_support::{Bytes, Erc20Token, EtherQuantity, Network};
+
+pub fn deploy_action(htlc_params: HtlcParams<Ethereum, Erc20Token>) -> ethereum::ContractDeploy {
+    htlc_params.into()
+}
+
+pub fn fund_action(
+    htlc_params: HtlcParams<Ethereum, Erc20Token>,
+    to_erc20_contract: ethereum_support::Address,
+    beta_htlc_location: ethereum_support::Address,
+) -> ethereum::SendTransaction {
+    let htlc = Erc20Htlc::from(htlc_params.clone());
```

Comment:
```suggestion
    let network = htlc_params.ledger.network;
    let htlc = Erc20Htlc::from(htlc_params);
    let gas_limit = Erc20Htlc::fund_tx_gas_limit();
```

I wonder if this would work.